pkg/apis/autoscaling/v1alpha1/pa_defaults.go

```diff
@@ -36,12 +37,13 @@ func defaultMetric(class string) string {
 
 func (r *PodAutoscaler) SetDefaults(ctx context.Context) {
 	r.Spec.SetDefaults(apis.WithinSpec(ctx))
+	config := config.FromContextOrDefaults(ctx)
 	if r.Annotations == nil {
 		r.Annotations = make(map[string]string)
 	}
 	if _, ok := r.Annotations[autoscaling.ClassAnnotationKey]; !ok {
-		// Default class to KPA.
-		r.Annotations[autoscaling.ClassAnnotationKey] = autoscaling.KPA
+		// Default class based on configmap setting (KPA if none specified).
+		r.Annotations[autoscaling.ClassAnnotationKey] = config.Autoscaler.PodAutoscalerClass
```

Comment:
Would we reduce the surface area of this change by a lot of we adopted the same defaulting behavior we do for ingress, which does the defaulting via the reconciler I think?

Reply:
Very possibly. It looks like we fairly intentionally did this with defaulting in https://github.com/knative/serving/pull/3103 though (see third bullet point). There is a fairly solid argument that it's better for all the defaulting to happen in one place, but also an argument that we already do some defaulting in reconcile anyway 🤔.

Reply:
(tbh arguably now that https://github.com/knative/serving/pull/6838 and https://github.com/knative/serving/pull/6822 are merged the surface area remaining isn't all that large any more 😅)